src/impls/$distinct/$distinct.js

```diff
@@ -0,0 +1,23 @@
+import { $async, $await } from '../../../generate/async.macro.cjs';
+
+import { $iterableCurry } from '../../internal/$iterable.js';
+
+$async;
+export function* $__distinct(source, selector) {
+  const set = new Set();
+  $await;
+  for (const value of source) {
+    const key = selector ? selector(value) : value;
+    if (set.has(key)) {
+      continue;
+    }
+
+    set.add(key);
+    yield value;
+  }
+}
+
+export const $distinct = $iterableCurry($__distinct, {
+  minArgs: 0,
+  maxArgs: 2,
```

Comment:
These numbers refer to the number of configuration arguments, i.e. those that are not iterables. You only have 1 arguments that is not an iterable.

Reply:
Sure. The currying code is a little bit confusing but yet quite smart.

Reply:
I should probably add some better comments to it.

Reply:
fixed in 64f50880549afd41939f95942a41006ffd65ce5d